remote-api/src/main/java/org/alfresco/rest/api/impl/NodesImpl.java

```diff
@@ -2971,6 +2985,14 @@ public Node upload(String parentFolderNodeId, FormData formData, Parameters para
                     renditionNames = getStringOrNull(field.getValue());
                     break;
 
+            case "versioningenabled":
+                    String versioningEnabledStringValue = getStringOrNull(field.getValue());
+                    if(null != versioningEnabledStringValue)
+                    {
+                        versioningEnabled = !versioningEnabledStringValue.equalsIgnoreCase("false");
```

Comment:
Any particular reason for testing for "false" and negating the result?

Wouldn't the following be easier to read and understand?

`versioningEnabled = versioningEnabledStringValue.equalsIgnoreCase("true");`

Reply:
I've decided to take this approach (still evaluating) because the default behavior for multipart upload is to enable auto-version for content, because of this, every request having an invalid value for `versioningEnabled` should be auto-versioned.

equalsIgnoreCase("true"); will return false for everything except the value `true`

Reply:
OK, as long as there's a legitimate reason, would be good to add that as a comment though for the next reader ;-)

Reply:
I`ve added a comment, thank you.